pkg/storaged/crypto-keyslots.jsx

```diff
@@ -253,6 +256,191 @@
+                      {
+                          opts: {
+                              t: 'ay',
+                              v: encode_filename(unparse_options(new_fsys_options))
+                          }
+                      })
+    ];
+    progress(cockpit.format(_('Adding "$0" to filesystem options'), option), null);
+    return block.UpdateConfigurationItem(fsys_config, new_fsys_config, { });
+}
+
+function ensure_crypto_option(steps, progress, client, block, option) {
+    const crypto_config = array_find(block.Configuration, function (c) { return c[0] == "crypttab" });
+    const crypto_options = crypto_config && parse_options(decode_filename(crypto_config[1].options.v));
+    if (!crypto_options || crypto_options.indexOf(option) >= 0)
+        return Promise.resolve();
+
+    const new_crypto_options = crypto_options.concat([option]);
+    progress(cockpit.format(_('Adding "$0" to encryption options'), option), null);
+    return edit_config(block, (config, commit) => {
+        config.options = { t: 'ay', v: encode_filename(unparse_options(new_crypto_options)) };
+        return commit();
+    });
+}
+
+function ensure_systemd_unit_enabled(steps, progress, name, package_name) {
+    progress(cockpit.format(_("Enabling $0"), name));
+    return cockpit.spawn(["systemctl", "is-enabled", name], { err: "message" })
+            .catch((err, output) => {
+                if (err && output == "" && package_name) {
+                // We assume that installing the package will enable the unit.
+                    return ensure_package_installed(steps, progress, package_name);
+                } else
+                    return cockpit.spawn(["systemctl", "enable", name],
+                                         { superuser: true, err: "message" });
+            });
+}
+
+function ensure_non_root_nbde_support(steps, progress, client, block) {
+    return ensure_systemd_unit_enabled(steps, progress, "remote-cryptsetup.target")
+            .then(() => ensure_systemd_unit_enabled(steps, progress, "clevis-luks-askpass.path", "clevis-systemd"))
+            .then(() => ensure_fstab_option(steps, progress, client, block, "_netdev"))
+            .then(() => ensure_crypto_option(steps, progress, client, block, "_netdev"));
+}
+
+function ensure_nbde_support(steps, progress, client, block) {
+    const cleartext = client.blocks_cleartext[block.path];
+    const crypto = client.blocks_crypto[block.path];
+    const fsys_config = (cleartext
+        ? array_find(cleartext.Configuration, function (c) { return c[0] == "fstab" })
+        : array_find(crypto.ChildConfiguration, function (c) { return c[0] == "fstab" }));
+    const dir = decode_filename(fsys_config[1].dir.v);
+
+    if (dir == "/") {
+        if (client.get_config("nbde_root_help", false)) {
+            steps.is_root = true;
+            return ensure_root_nbde_support(steps, progress);
+        } else
+            return Promise.resolve();
+    } else
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17796-20221109-135428-ee4e0d98-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)